src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborReader.String.cs

```diff
@@ -49,30 +82,91 @@ public bool TryReadByteString(Span<byte> destination, out int bytesWritten)
             {
                 if (_isConformanceLevelCheckEnabled && CborConformanceLevelHelpers.RequiresDefiniteLengthItems(ConformanceLevel))
                 {
-                    throw new FormatException("Indefinite-length items not supported under the current conformance level.");
+                    throw new FormatException(SR.Format(SR.Cbor_ConformanceLevel_IndefiniteLengthItemsNotSupported, ConformanceLevel));
                 }
 
-                return TryReadChunkedByteStringConcatenated(destination, out bytesWritten);
+                return TryReadIndefiniteLengthByteStringConcatenated(destination, out bytesWritten);
             }
 
-            int length = checked((int)ReadUnsignedInteger(_buffer.Span, header, out int additionalBytes));
-            EnsureBuffer(1 + additionalBytes + length);
+            ReadOnlySpan<byte> buffer = GetRemainingBytes();
+            int length = DecodeDefiniteLength(header, buffer, out int bytesRead);
+            EnsureReadCapacity(bytesRead + length);
 
             if (length > destination.Length)
             {
                 bytesWritten = 0;
                 return false;
             }
 
-            _buffer.Span.Slice(1 + additionalBytes, length).CopyTo(destination);
-            AdvanceBuffer(1 + additionalBytes + length);
+            buffer.Slice(bytesRead, length).CopyTo(destination);
+            AdvanceBuffer(bytesRead + length);
             AdvanceDataItemCounters();
 
             bytesWritten = length;
             return true;
         }
 
-        // Implements major type 3 decoding per https://tools.ietf.org/html/rfc7049#section-2.1
+        /// <summary>
+        ///   Reads the next data item as the start of an indefinite-length byte string (major type 2)
+        /// </summary>
+        /// <returns>The decoded byte array.</returns>
+        /// <exception cref="InvalidOperationException">
+        ///   the next data item does not have the correct major type.
+        /// </exception>
```

Comment:
Missing the exception cause of "the next data item has a definite-length encoding"

Reply:
I presume you mean indefinite-length? This will either be concatenated, or a throw a FormatException when the conformance level does not support indefinite-length encodings. The second error is grouped in the `the next value uses a CBOR encoding that is not valid under the current conformance level.` category.

Reply:
I'm just reading the code: 

```C#
            if (header.AdditionalInfo != CborAdditionalInfo.IndefiniteLength)
            {
                throw new InvalidOperationException(SR.Cbor_Reader_NotIndefiniteLengthString);
            }
```

I read that as "throws InvalidOperationException if the next data item has a definite-length encoding".

Reply:
Ah, right. I thought this was a different method. Fixed.